pkg/receive/handler_test.go

```diff
@@ -1841,8 +1855,6 @@ func TestHandlerFlippingHashrings(t *testing.T) {
 }
 
 func TestIngestorRestart(t *testing.T) {
-	// TODO: fix this test. It has a data race.
-	t.Skip("Skipping this test case temporarily due to a data race")
```

Comment:
fyi @hczhu-db, fixed the data race of the test

Reply:
nice work